tests/Doctrine/Tests/ORM/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -451,6 +454,70 @@ public function testInheritsIdGeneratorMappingFromEmbeddable()
 
         $this->assertTrue($userMetadata->isIdGeneratorIdentity());
     }
+
+    public function testPropertyOrderNoCache()
+    {
+        $cmf = $this->createPropertyOrderFactory();
+
+        /** @var ClassMetadataInfo $adminMetadata */
+        $adminMetadata = $cmf->getMetadataFor(PropertyOrder\Admin::class);
+        $this->assertSame(
+            ['id', 'name', 'group', 'email'],
+            array_keys($adminMetadata->reflFields)
+        );
+
+        /** @var ClassMetadataInfo $userMetadata */
+        $userMetadata = $cmf->getMetadataFor(PropertyOrder\User::class);
+        $this->assertSame(
+            ['id', 'name', 'group'],
+            array_keys($userMetadata->reflFields)
+        );
+    }
+
+    public function testPropertyOrderWithCache()
+    {
+        $noCacheCmf = $this->createPropertyOrderFactory();
+        $cache      = new ArrayCache();
```

Comment:
That's why the `reflFields` property is reset manually. I meant to add a comment about that.

Alternatively could (de)serialize manually, or use another cache implementation.